strawberry/annotation.py

```diff
@@ -187,7 +187,7 @@ def create_enum(self, evaled_type: Any) -> EnumDefinition:
         try:
             return evaled_type._enum_definition
         except AttributeError:
-            raise NotAStrawberryEnumError(evaled_type)
```

Comment:
How do we avoid duplicates here, in case the same enum is used multiple times?

This (which is called by `strawberry_enum` will create a new enum definition every time and override the old).

https://github.com/nrbnlulu/strawberry/blob/73ce3251d64ee63c2e36eecfde8f5eef927889a9/strawberry/types/enum.py#L144-L150

How about we add an extra check in there? Otherwise we'd need a registry to check on-the-fly converted enums. Additionally, a test for this case (same non-strawberry enum used twice) would be great!

Reply:
Yeah I figured that. I can't find a proper solution because strawberry doesn't have a registry. and I'm not sure adding a global one just for enums doesn't make sense :thinking:

Reply:
can't we do 

```python
def _process_enum(
    cls: EnumType,
    name: Optional[str] = None,
    description: Optional[str] = None,
    directives: Iterable[object] = (),
) -> EnumType:

  if cls.__enum_definition__:
   return cls.__enum_definition__
```

Reply:
ah this should be possible :+1:

Reply:
Actually the current implementation takes care of it without me doing anything
https://github.com/nrbnlulu/strawberry/blob/608afeb098b7cb4a04e631e21f84c201271727fa/strawberry/types/enum.py#L144

https://github.com/nrbnlulu/strawberry/blob/608afeb098b7cb4a04e631e21f84c201271727fa/strawberry/annotation.py#L190